src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -46,22 +36,24 @@ public ActiveNotifier(SlackNotifier notifier, BuildListener listener) {
 
     private SlackService getSlack(AbstractBuild r) {
         AbstractProject<?, ?> project = r.getProject();
-        String projectRoom = Util.fixEmpty(project.getProperty(SlackNotifier.SlackJobProperty.class).getRoom());
-        String teamDomain = Util.fixEmpty(project.getProperty(SlackNotifier.SlackJobProperty.class).getTeamDomain());
-        String token = Util.fixEmpty(project.getProperty(SlackNotifier.SlackJobProperty.class).getToken());
+        String projectRoom = Util.fixEmpty(project.getProperty(SlackJobProperty.class).getRoom());
+        String teamDomain = Util.fixEmpty(project.getProperty(SlackJobProperty.class).getTeamDomain());
+        String token = Util.fixEmpty(project.getProperty(SlackJobProperty.class).getToken());
+
+        // Support for direct messaging
+        String directMessage = Util.fixEmpty(project.getProperty(SlackJobProperty.class).getSendDirectMessage());
 
         EnvVars env = null;
         try {
             env = r.getEnvironment(listener);
         } catch (Exception e) {
             listener.getLogger().println("Error retrieving environment vars: " + e.getMessage());
-            env = new EnvVars();
         }
         teamDomain = env.expand(teamDomain);
```

Comment:
By removing line 58, line 60 will throw a NullPointerException if there's any problem retrieving the environment.